src/Form/RenderFileTab.tsx

```diff
@@ -82,16 +111,34 @@ export function RenderBasicFileTab({ step, splitSchema, setSplitSchema }: Render
     setStepState(splitSchema, setSplitSchema, step, { ...state, binary: e.target.files[0] })
   }
 
+  const hasUploadType = useMemo(() => !!buildOptionsStep.state.uploadType, [buildOptionsStep])
+
   return (
     <Box sx={{ pb: 4, pt: 4 }}>
-      <Stack direction='row' spacing={2} alignItems='center'>
-        <FileInput label='Select Code' file={code} onChange={handleCodeChange} accepts='.zip' />
-        <FileInput label='Select Binary' file={binary} onChange={handleBinaryChange} accepts='.zip' />
-      </Stack>
+      {(!hasUploadType ||
+        (buildOptionsStep !== undefined && buildOptionsStep.state.uploadType === ModelUploadType.Zip)) && (
+        <Stack direction='row' spacing={2} alignItems='center'>
+          <FileInput label='Select Code' file={code} onChange={handleCodeChange} accepts='.zip' />
+          <FileInput label='Select Binary' file={binary} onChange={handleBinaryChange} accepts='.zip' />
+        </Stack>
+      )}
+      {hasUploadType && buildOptionsStep.state.uploadType === ModelUploadType.ModelCard && (
+        <Typography sx={{ pt: 2, pb: 2 }}>Uploading a model card without any code or binary files</Typography>
+      )}
     </Box>
   )
 }
 
 export function BasicFileTabComplete(step: Step) {
```

Comment:
This returns a boolean, not a component. Consider renaming to `basicFileTabComplete`